adapters/AppLovin/AppLovinAdapter/GADMAdapterAppLovin.m

```diff
@@ -193,15 +186,204 @@ - (ALInterstitialAd *)interstitial {
+    else
+    {
+        CGSize frameSize = size.size;
+        if ( CGRectGetWidth([UIScreen mainScreen].bounds) == frameSize.width )
+        {
+            CGFloat frameHeight = frameSize.height;
+            if ( frameHeight == CGSizeFromGADAdSize(kGADAdSizeBanner).height )
+            {
+                return [ALAdSize sizeBanner];
+            }
+            else if ( frameHeight == CGSizeFromGADAdSize(kGADAdSizeMediumRectangle).height )
+            {
+                return [ALAdSize sizeMRec];
+            }
+            else if ( frameHeight == CGSizeFromGADAdSize(kGADAdSizeLeaderboard).height )
+            {
+                return [ALAdSize sizeLeader];
+            }
+        }
+    }
+    
+    [self log: @"Unable to retrieve AppLovin size from GADAdSize: %@", NSStringFromGADAdSize(size)];
+    
+    return nil;
+}
+
+#pragma mark - Utility Methods
+
+- (void)log:(NSString *)format, ...
+{
+    if ( loggingEnabled )
+    {
+        va_list valist;
+        va_start(valist, format);
+        NSString *message = [[NSString alloc] initWithFormat: format arguments: valist];
+        va_end(valist);
+        
+        NSLog(@"AppLovinAdapter: %@", message);
+    }
+}
+
+@end
+
+@implementation AppLovinAdMobBannerDelegate
+
+#pragma mark - Initialization
+
+- (instancetype)initWithParentAdapter:(GADMAdapterAppLovin *)parentAdapter
+{
+    self = [super init];
+    if ( self )
+    {
+        self.parentAdapter = parentAdapter;
+    }
+    return self;
+}
+
+#pragma mark - AppLovin Ad Load Delegate
+
+- (void)adService:(ALAdService *)adService didLoadAd:(ALAd *)ad
+{
+    [self.parentAdapter log: @"Banner did load ad: %@", ad.adIdNumber];
+    
+    NSString *placement = [GADMAdapterAppLovinUtils placementFromCredentials: self.parentAdapter.connector.credentials];
+    [self.parentAdapter.adView render: ad overPlacement: placement];
+    
+    [self.parentAdapter.connector adapter: self.parentAdapter didReceiveAdView: self.parentAdapter.adView];
+}
+
+- (void)adService:(ALAdService *)adService didFailToLoadAdWithError:(int)code
+{
+    [self.parentAdapter log: @"Banner failed to load with error: %d", code];
+    
+    NSError *error = [NSError errorWithDomain: kGADMAdapterAppLovinErrorDomain
+                                         code: [GADMAdapterAppLovinUtils toAdMobErrorCode: code]
+                                     userInfo: nil];
+    [self.parentAdapter.connector adapter: self.parentAdapter didFailAd: error];
+}
+
+#pragma mark - Ad Display Delegate
+
+- (void)ad:(ALAd *)ad wasDisplayedIn:(UIView *)view
+{
+    [self.parentAdapter log: @"Banner displayed"];
+}
+
+- (void)ad:(ALAd *)ad wasHiddenIn:(UIView *)view
+{
+    [self.parentAdapter log: @"Banner dismissed"];
+}
+
+- (void)ad:(ALAd *)ad wasClickedIn:(UIView *)view
+{
+    [self.parentAdapter log: @"Banner clicked"];
+    
+    [self.parentAdapter.connector adapterDidGetAdClick: self.parentAdapter];
+    
+    if ( !AD_VIEW_EVENT_DELEGATE_AVAILABLE )
```

Comment:
Seems like this is always evaluating to `true` and I was getting double will leave app callbacks.

Reply:
Fixed an issue with the macro expansion, sorry about that.